utils/parametric-components.stanza

```diff
@@ -0,0 +1,537 @@
+  import math
+
+  import jitx
+  import jitx/commands
+  import esir/ir
+
+
+  import ocdb/utils/bundles
+  import ocdb/utils/landpatterns
+  import ocdb/utils/generator-utils
+  import ocdb/utils/db-parts
+  import ocdb/utils/design-vars
+  import ocdb/utils/symbols
+  import ocdb/utils/box-symbol
+  import ocdb/utils/property-structs
+  import ocdb/utils/symbol-utils
+  import ocdb/utils/checks
+  import ocdb/utils/parts
+  import ocdb/utils/generic-components
+
+
+public pcb-enum /ocdb/utils/parametric-components/ResistorApplication :
+  ResistorPull
+  ResistorPrecision
+  ResistorTerm
+  ResistorSense
+  ResistorPower
+;  ; application:ResistorApplication
+
+public pcb-enum /ocdb/utils/parametric-components/CapacitorApplication :
+  CapBypass
+  CapFilter
+  CapPrecision
+  CapLoad
+
+; ; ========================================================
+; ; ========== Generic passive components ==================
+; ; ========================================================
+
+public pcb-component dummy-resistor (res:Double|Toleranced orig:Double|Toleranced|False = false):
+  port p : pin[[1 2]]
+  val sym = resistor-sym(ResistorStd)
+  symbol = sym(p[1] => sym.p[1], p[2] => sym.p[2])
+  val pkg = ipc-two-pin-landpattern("0402")
+  landpattern = pkg(p[1] => pkg.p[1], p[2] => pkg.p[2])
+  reference-prefix = "R"
+  property(self.resistance) = res
+  property(self.resistor) = true
+  match(orig:Double|Toleranced) :
+    property(self.orig-resistance) = orig
+  else :
+    property(self.orig-resistance) = res
+  ; spice :
+  ;   "[R] <p[1]> <p[2]> <res>"
+
+public pcb-component dummy-capacitor (cap:Double|Toleranced orig:Double|Toleranced|False = false):
+  port p : pin[[1 2]]
+  val sym = capacitor-sym(CapacitorNonPolarized)
+  symbol = sym(p[1] => sym.p[1], p[2] => sym.p[2])
+  val pkg = ipc-two-pin-landpattern("0402")
+  landpattern = pkg(p[1] => pkg.p[1], p[2] => pkg.p[2])
+  reference-prefix = "C"
+  property(self.capacitor) = true
+  property(self.type) = "ceramic"
+  property(self.capacitance) = cap
+  match(orig:Double|Toleranced) :
+    property(self.orig-capacitance) = orig
+  else :
+    property(self.orig-capacitance) = cap
+  ; spice :
+  ;   "[C] <p[1]> <p[2]> <cap>"
+
+public pcb-component dummy-inductor (ind:Double|Toleranced orig:Double|Toleranced|False = false):
+  port p : pin[[1 2]]
+  val sym = inductor-sym(InductorStd)
+  symbol = sym(p[1] => sym.p[1], p[2] => sym.p[2])
+  val pkg = ipc-two-pin-landpattern("0402")
+  landpattern = pkg(p[1] => pkg.p[1], p[2] => pkg.p[2])
+  reference-prefix = "L"
+  property(self.inductor) = true
+  property(self.inductance) = ind
+  match(orig:Double|Toleranced) :
+    property(self.orig-inductance) = orig
+  else :
+    property(self.orig-inductance) = ind
+  ; spice :
+  ;   "[L] <p[1]> <p[2]> <ind>"
+
+defn smd-query-properties () -> Tuple<KeyValue<String, ?>> :
+  ["mounting" => "smd"
+   "case" => get-valid-pkg-list()
+   "minimum_quantity" => 1
+   "min-stock" => 5 * DESIGN-QUANTITY]
```

Comment:
You may want to deduplicate with the slightly different function in generic-components or not use the same function name
```
defn smd-query-properties () -> Tuple<KeyValue<String, ?>> :
  ["mounting" => "smd"
   "case" => get-valid-pkg-list()
   "min-stock" => 5 * DESIGN-QUANTITY]
```

Reply:
Good suggestion. I made the one in generic-components public and deleted this defn.